### ISSUE TITLE
Investigate Failing tests on "Latest"

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#739]
http://services.typefox.io/open-source/jenkins/job/xtext-eclipse/job/master/47/#showFailuresLink

    org.eclipse.xtext.ui.tests.editor.hover.AbstractCompositeHoverTest.testMultipleHovers
    org.eclipse.xtext.ui.tests.editor.hover.AbstractCompositeHoverTest.testNullHoverList
    org.eclipse.xtext.ui.tests.editor.hover.AbstractCompositeHoverTest.testSingleHover
    org.eclipse.xtext.ui.tests.editor.hover.AbstractCompositeHoverTest.testEmptyHoverList
    org.eclipse.xtext.ui.tests.editor.hover.AbstractProblemHoverTest.testAnnotations
    org.eclipse.xtext.ui.tests.editor.hover.AbstractProblemHoverTest.testMultipleAnnotationTypes
    org.eclipse.xtext.ui.tests.editor.hover.AbstractProblemHoverTest.testAnnotationOnMultipleLines
    org.eclipse.xtext.ui.tests.editor.hover.AnnotationWithQuickFixesHoverTest.testAnnotations
    org.eclipse.xtext.ui.tests.editor.hover.AnnotationWithQuickFixesHoverTest.testNoAnnotations
    org.eclipse.xtext.ui.tests.editor.hover.DispatchingEObjectTextHoverTest.testNames
    org.eclipse.xtext.ui.tests.editor.hover.DispatchingEObjectTextHoverTest.testCrossReferences
    org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.testAnnotations
    org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.testBug357516_error
    org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.testBug357516_warning
    org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.testBug357516_info
    org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.testBug357516_bookmark
    org.eclipse.xtext.ui.tests.editor.hover.html.XtextElementLinksTest.testDefaultScheme
    org.eclipse.xtext.ui.tests.editor.hover.html.XtextElementLinksTest.testOPEN_LINK_SCHEME
    org.eclipse.xtext.ui.tests.editor.hover.html.XtextElementLinksTest.testXTEXTDOC_SCHEME
    org.eclipse.xtext.ui.tests.editor.hover.html.XtextElementLinksTest.testXTEXTDOC_VIEW_SCHEME
    org.eclipse.xtext.ui.tests.editor.outline.Bug326812Test.testCloseTwoEditorsForSameLanguage
    org.eclipse.xtext.ui.tests.editor.outline.Bug379500Test.testOutlineContribution
    org.eclipse.xtext.ui.tests.editor.outline.FilterContributionTest.testFilteringPreservesExpansionAndSelection
    org.eclipse.xtext.ui.tests.editor.outline.FilterContributionTest.testFilterOutline
    org.eclipse.xtext.ui.tests.editor.outline.LinkWithEditorTest.testTreeToText
    org.eclipse.xtext.ui.tests.editor.outline.LinkWithEditorTest.testTreeToText
    org.eclipse.xtext.ui.tests.editor.outline.LinkWithEditorTest.testTextToTree
    org.eclipse.xtext.ui.tests.editor.outline.LinkWithEditorTest.testTextToTree
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testAppend
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testAppend
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testInsertBefore
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testInsertBefore
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testChangeOtherNode
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testChangeOtherNode
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testChangeSelection
    org.eclipse.xtext.ui.tests.editor.outline.PreserveExpansionAndSelectionTest.testChangeSelection
    org.eclipse.xtext.ui.tests.editor.outline.SortContributionTest.testSort
    org.eclipse.xtext.ui.tests.editor.outline.SortContributionTest.testSort
    org.eclipse.xtext.ui.tests.editor.outline.SortContributionTest.testNoSort
    org.eclipse.xtext.ui.tests.editor.outline.SortContributionTest.testNoSort
    org.eclipse.xtext.ui.tests.editor.outline.SortContributionTest.testSortingPreservesSelectionAndExpansion
    org.eclipse.xtext.ui.tests.editor.outline.SortContributionTest.testSortingPreservesSelectionAndExpansion
    org.eclipse.xtext.ui.tests.editor.quickfix.LinkingErrorTest.testSemanticIssueResolution
    org.eclipse.xtext.ui.tests.editor.quickfix.LinkingErrorTest.testBug361509
    org.eclipse.xtext.ui.tests.editor.quickfix.LinkingErrorTest.testQuickfixTurnaround
    org.eclipse.xtext.ui.tests.editor.quickfix.SpellingQuickfixTest.testSpellingQuickfixInSlComment
    org.eclipse.xtext.ui.tests.editor.quickfix.SpellingQuickfixTest.testSpellingQuickfixInMlComment
    org.eclipse.xtext.ui.tests.editor.quickfix.SpellingQuickfixTest.testSpellingQuickfixInString

**Comment by cdietrich** [comment]
somehow org.eclipse.xtext.ui.testing.AbstractEditorTest.openEditor(IFile)
does no longer seem to work. i wonder what was changed in platform again
@iloveeclipse do you have any hints?

**Comment by iloveeclipse** [comment]
What is "latest"? 4.9 HEAD?

**Comment by iloveeclipse** [comment]
Christian, can you configure Jenkins job to capture workspace logs please?

**Comment by cdietrich** [comment]
i will have a look an how to do that.

**Comment by cdietrich** [comment]
http://services.typefox.io/open-source/jenkins/job/xtext-eclipse/job/master/47/execution/node/3/ws/org.eclipse.xtext.ui.tests/target/work/data/.metadata/.log

**Comment by iloveeclipse** [comment]
At the very bottom of the job config you have to enter regex to search for, we have this in platform UI:

Files to archive: `tests/*/target/work/data/.metadata/.log, **/hs_err_pid*.log`

**Comment by iloveeclipse** [comment]
You have there tons of errors. Which one are new on 4.9?

**Comment by iloveeclipse** [comment]
I guess this is the problem:
`java.lang.NullPointerException
	at org.eclipse.xtext.ui.containers.AbstractJavaProjectsState.isAffectingPackageFragmentRoots(AbstractJavaProjectsState.java:65)
	at org.eclipse.xtext.ui.containers.AbstractJavaProjectsState.isAffectingPackageFragmentRoots(AbstractJavaProjectsState.java:79)
	at org.eclipse.xtext.ui.containers.AbstractJavaProjectsState.isAffectingPackageFragmentRoots(AbstractJavaProjectsState.java:63)
	at org.eclipse.xtext.ui.containers.AbstractJavaProjectsState.elementChanged(AbstractJavaProjectsState.java:50)
	at org.eclipse.jdt.internal.core.DeltaProcessor$3.run(DeltaProcessor.java:1735)

**Comment by cdietrich** [comment]
i wonder why tycho surefire does not give a complete trace

**Comment by iloveeclipse** [comment]
`@Inject
	private JavaProjectClasspathChangeAnalyzer javaProjectClasspathChangeAnalyzer;` does not work?

**Comment by cdietrich** [comment]
all tests are on 4.9.
they seem to work against photon and older.

**Comment by iloveeclipse** [comment]
```!ENTRY org.eclipse.jdt.core 4 4 2018-07-09 19:42:44.497
!MESSAGE Error while creating a link for external folder :/var/jenkins_home/workspace/xtext-eclipse_master-OX2L6OFRZ75ZC7A3K3BJAQVU3V3QABS2LI4URYKVIRLWPPW5BHSQ/org.eclipse.xtext.ui.tests/target/work/data/.org.eclipse.jdt.core.external.folders/externalFolder
!STACK 1
org.eclipse.core.internal.resources.ResourceException: Cannot create linked resource '/.org.eclipse.jdt.core.external.folders/.link6'.  The parent resource is not accessible.
	at org.eclipse.core.internal.resources.Resource.assertLinkRequirements(Resource.java:167)
	at org.eclipse.core.internal.resources.Resource.createLink(Resource.java:629)
	at org.eclipse.core.internal.resources.Resource.createLink(Resource.java:605)```

Too long parent path may be?

**Comment by cdietrich** [comment]
dont see these npe if i run the test only
=> need to remote debug

**Comment by cdietrich** [comment]
stacktrace of the npe

```
	at org.eclipse.ui.internal.PartSite.getShell(PartSite.java:474)
	at org.eclipse.ui.texteditor.AbstractTextEditor.close(AbstractTextEditor.java:4295)
	at org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.tearDown(ProblemHoverTest.java:84)

```

**Comment by cdietrich** [comment]
no the injection is a different issue.

**Comment by cdietrich** [comment]
is it possible it is related to this change: https://github.com/eclipse/eclipse.platform.ui/commit/8cbaa58e8bff0bc61963dea8af5329e9095f1834#diff-0b3b184e168187435b213a9079f692a3

**Comment by iloveeclipse** [comment]
The NPE is  from me :-) 
https://bugs.eclipse.org/bugs/show_bug.cgi?id=536677
The code tries to use already disposed part site, and the context is null.
Do you have the full stack?

**Comment by cdietrich** [comment]
```
	at org.eclipse.ui.internal.PartSite.getShell(PartSite.java:474)
	at org.eclipse.ui.texteditor.AbstractTextEditor.close(AbstractTextEditor.java:4295)
	at org.eclipse.xtext.ui.tests.editor.hover.ProblemHoverTest.tearDown(ProblemHoverTest.java
```

**Comment by cdietrich** [comment]
(e4Context is null)

**Comment by cdietrich** [comment]
Fixed in 2.15
